tasks/src/tasks/sort.rs

```diff
@@ -0,0 +1,84 @@
+use std::num::NonZeroU32;
+
+use imap_types::{
+    command::CommandBody,
+    core::{Charset, Vec1},
+    extensions::sort::SortCriterion,
+    response::{Data, StatusBody, StatusKind},
+    search::SearchKey,
+};
+
+use super::TaskError;
+use crate::Task;
+
+#[derive(Clone, Debug)]
+pub struct SortTask {
+    sort_criteria: Vec1<SortCriterion>,
+    charset: Charset<'static>,
+    search_criteria: Vec1<SearchKey<'static>>,
+    uid: bool,
+    output: Vec<NonZeroU32>,
+}
+
+impl Task for SortTask {
+    type Output = Result<Vec<NonZeroU32>, TaskError>;
+
+    fn command_body(&self) -> CommandBody<'static> {
+        CommandBody::Sort {
+            sort_criteria: self.sort_criteria.clone(),
+            charset: self.charset.clone(),
+            search_criteria: self.search_criteria.clone(),
+            uid: self.uid,
+        }
+    }
+
+    fn process_data(&mut self, data: Data<'static>) -> Option<Data<'static>> {
+        if let Data::Sort(ids) = data {
+            self.output = ids;
+            None
+        } else {
+            Some(data)
+        }
+    }
+
+    fn process_tagged(self, status_body: StatusBody<'static>) -> Self::Output {
+        match status_body.kind {
+            StatusKind::Ok => Ok(self.output),
+            StatusKind::No => Err(TaskError::UnexpectedNoResponse(status_body)),
+            StatusKind::Bad => Err(TaskError::UnexpectedBadResponse(status_body)),
+        }
+    }
+}
+
+impl SortTask {
```

Comment:
Thought: Maybe we want to unify all tasks later to use the builder pattern. Simple changes we can discuss later. Let's first try to get all tasks in :-)

Reply:
Sure, I like the idea :+1: